AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine.swift

```diff
@@ -258,6 +284,51 @@ class RemoteSyncEngine: RemoteSyncEngineBehavior {
         group.wait()
         onComplete()
     }
+
```

Comment:
Considering putting all of the retryability stuff into a category RemoteSyncEngine+Retryability.swift -- thoughts?

Reply:
I like the idea, especially since it starts to enforce the idea of Retriability as a strategy we can apply to a number of uses, not just RemoteSyncEngine. Maybe in a future PR, so as not to bloat this one?